File: Statista/Statista.Domain/Entities/Survey/AdminGroup.cs
using Statista.Domain.Entities;

public class AdminGroup
{
    public Guid SurveyId { get; set; }
    public Survey Survey { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; }
    public Guid RoleId { get; set; }
    public Classifier Role { get; set; }
}

File: statistika_mobile/lib/features/survey/domain/model/admin_group/admin_group.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:statistika_mobile/features/authorization/domain/model/user.dart';

import '../../../../../core/model/classifier.dart';

part 'admin_group.g.dart';

@JsonSerializable()
class AdminGroup {
  AdminGroup({
    required this.surveyId,
    required this.userId,
    required this.user,
    required this.roleId,
    required this.role,
  });

  final String surveyId;
  final String? userId;
  final User? user;
  final String? roleId;
  final Classifier? role;

  factory AdminGroup.fromJson(Map<String, dynamic> json) =>
      _$AdminGroupFromJson(json);

  Map<String, dynamic> toJson() => _$AdminGroupToJson(this);
}
